### DISCUSSION TITLE
Tiled 1.8 Release Candidate

### DISCUSSION THREAD

**Discussion by bjorn** [mapeditor/tiled#3253] [General]
This is tagged `v1.7.91`, but it is actually Tiled 1.8.0 RC. Please help with testing and report issues either here, by opening issues or on the [Tiled Discord](https://discord.gg/39wbTv7) server. Thanks!

## Changes since Tiled 1.8 Beta

* Added cut/copy/paste actions to Tile Animation Editor
* Improved undo behavior by merging sequential edits to the same property (#3103)
* Added import and export actions for property types
* Added confirmation before removing types, values or members
* Avoid circular references in property types on loading
* Avoid name collisions in types and class members in the Property Type Editor
* Report an error when a property type can't be found
* Limit enums with values as flags to max 32 values
* Separated the X and Y components of the major grid option (#3208)
* Added automatic fading out of the grid when zooming out a lot
* AutoMapping: Allow setting a rule map as project rules file (#3221)
* AutoMapping: `regions` layer now always defines input and output regions
* Show the read error when using --export-map/tileset
* Scripting: Added tiled.propertyValue, for creating values of custom types
* Scripting: Extended the terrain related API (#2663)
* Scripting: Added Tool.usesWangSets
* Scripting: tiled.activeAsset can be assigned asset created in script (#3160)
* Scripting: Fixed possible crash after creating tilesets from script (#3229)
* Fixed adjusting of tile types when tileset width changed (by Albert Vaca Cintora, #3237)
* Fixed missing Qt translations for Linux AppImage
* Fixed a possible crash when switching between different property types
* Fixed minimap viewport position when layers are offset (#3211)
* Fixed "Highlight Current Layer" getting confused (#3223)
* Fixed saving broken references to files loaded using "ext:" prefix (#3185)
* Fixed performance issue in Project view related to file icons
* TMX format: Include propertytype attribute for class members (#3230)
* Windows: The installers are signed again (1.8 Beta wasn't) and the WinXP build is included
* macOS: The Tiled.app is signed and notarized again (1.8 Beta wasn't)

<hr /><em>This discussion was created from the release <a href='https://github.com/mapeditor/tiled/releases/tag/v1.7.91'>Tiled 1.8 Release Candidate</a>.</em>

**Comment by user72356** [comment]
Hey @bjorn, have you had a chance to update the TypeScript definitions for the new 1.8 release? The @mapeditor/tiled-api package. 

There was this issue here: https://github.com/mapeditor/tiled/commit/c56e9861bab150f9d9f3baf559d312a7f9745a80#commitcomment-60662151. I made sure I was on the latest version of the @mapeditor/tiled-api package and that issue is still present. And the new 1.8 features are still missing from it. I don't know if it's a manual or automated process on your end, so I'm not making a big deal of it. Just a reminder for us Typescript extension makers :-)

**Reply by bjorn** [reply]
Thanks for the reminder! I've just published [`tiled-api`](https://www.npmjs.com/package/@mapeditor/tiled-api) 1.8!

**Reply by user72356** [reply]
Thanks! Just found an issue with it. Looks like you forgot the `: void;` after `public removeObjects(objects: MapObject[])`

```js
node_modules/@mapeditor/tiled-api/index.d.ts:2133:10 - error TS7010: 'removeObjects', which lacks return-type annotation, implicitly has an 'any' return type.

2133   public removeObjects(objects : MapObject[]);
```

**Reply by bjorn** [reply]
Hmm, I generated the docs with it and it didn't even warn about this. Seems I should look into enabling strict mode checks somewhere.

**Reply by bjorn** [reply]
I've released `tiled-api` 1.8.1, which now uses strict mode.

**Reply by user72356** [reply]
Hi @bjorn, I found another one:

```js
/**
 * Currently selected asset, or `null` if no file is open.
 * Can be assigned any open asset in order to change the active asset.
 */
export let activeAsset: Asset;
```

Should be:

```js
export let activeAsset: Asset | null;
```

**Reply by bjorn** [reply]
@user72356 Alright, I've addressed that in 16773a572b0f00151792df55627e5c5841b93b42. Do you need this released urgently?

**Reply by user72356** [reply]
Thank you, not urgent, the compiler gives an error but it still compiles.

**Reply by imartdecor1** [reply]
Grout King uses advanced tools and techniques to remove dirt, stains, and grime from grout lines and tiles, restoring floors and walls to immaculate condition. https://groutking.ae/

**Comment by imartdecor1** [comment]
Grout King uses advanced tools and techniques to remove dirt, stains, and grime from grout lines and tiles, restoring floors and walls to immaculate condition. https://groutking.ae/

**Comment by readymedicine11** [comment]
Buy natural Chaunsa, Sindhri, and Anwar Ratol mangoes online from Multanpoint at affordable prices. [buy online chaunsa mango](https://multanpoint.com/)
